chef-config/lib/chef-config/config.rb

```diff
@@ -107,12 +107,19 @@ def self.apply_extra_config_options(extra_config_options)
           # Split including whitespace if someone does truly odd like
           # --config-option "foo = bar"
           key, value = option.split(/\s*=\s*/, 2)
+
           # Call to_sym because Chef::Config expects only symbol keys. Also
           # runs a simple parse on the string for some common types.
           memo[key.to_sym] = YAML.safe_load(value)
           memo
         end
-        merge!(extra_parsed_options)
+        set_extra_config_options(extra_parsed_options)
+      end
+    end
+
+    def self.set_extra_config_options(extra_parsed_options)
```

Comment:
You might want to include a comment here:  "# We use :[]= assignment here to not bypass any coercions that happen via mixlib-config writes_value callbacks"